client/virtual_channel.go

```diff
@@ -97,6 +98,50 @@ func (c *Client) handleVirtualChannelFundingProposal(
 	c.acceptProposal(responder)
 }
 
+func (c *Channel) watchVirtual() error {
+	log := c.Log().WithField("proc", fmt.Sprintf("virtual channel watcher %v", c.ID()))
+	defer log.Info("Watcher returned.")
+
+	// Subscribe to state changes
+	ctx := c.Ctx()
+	sub, err := c.adjudicator.Subscribe(ctx, c.Params())
+	if err != nil {
+		return errors.WithMessage(err, "subscribing to adjudicator state changes")
+	}
+	defer func() {
+		if err := sub.Close(); err != nil {
+			log.Warn(err)
+		}
+	}()
+
```

Comment:
The subscription receives the channel context and should therefore close anyway, when the channel is closed, right?

Reply:
I think the Adjudicator sub is a bit unintuitive in its usage.  
You can test it by placing a *panic* in L114 in the *defer* after the *close* and see that it is never executed.

Reply:
It doesn't panic because channel closing is not included in the test. Furthermore, the mock subscription was not designed to respect the context.

I now changed the mock subscription to respect its context when waiting for the next event. In the happy virtual channel test you can see that the watcher returns as expected. In the dispute test, we would need to call close manually for all the settled channels. I am a bit reluctant to do that as it adds a lot of code and has not much value in my opinion. We should definitely consider closing channels automatically when they are settled. I will create an issue for that.